tests/org.eclipse.reddeer.eclipse.test/src/org/eclipse/reddeer/eclipse/test/jdt/ui/wizards/NewInterfaceCreationWizardTest.java

```diff
@@ -0,0 +1,178 @@
+		}
+
+		assertTrue(
+				"List of interfaces obtained using function getExtendedInterfaces() has to be the same as list interfaces, that has been set.",
+				interfacePage.getExtendedInterfaces().equals(extendedIterfaceList));
+
+		for (String extendedInterfaceName : extendedIterfaceList) {
+			interfacePage.removeExtendedInterface(extendedInterfaceName);
+		}
+
+		// toggle generate comments checkbox
+		interfacePage.toggleGenerateComments(true);
+		interfacePage.toggleGenerateComments(false);
+
+		// test values in fields
+		assertTrue(
+				"Interface name obtained from the input using getName() function is not the same as name set using setName().",
+				interfacePage.getName().equals(INTERFACE_NAME));
+		assertTrue(
+				"Package name obtained from the input using getPackage() function is not the same as name set using setPackage().",
+				interfacePage.getPackage().equals(PACKAGE));
+		assertTrue(
+				"Source folder name obtained from the input using getSourceFolder() function is not the same as name set using setSourceFolder().",
+				interfacePage.getSourceFolder().equals(SOURCE_FOLDER_NAME));
+
+		// finish the wizard
+		interfaceDialog.finish();
+	}
+
+	@Test
+	public void testAddingAndDeletingNonexistingAndNotFullySpecifiedInterfaces() {
+		// open the wizard
+		NewInterfaceCreationWizard interfaceDialog = new NewInterfaceCreationWizard();
+		interfaceDialog.open();
+		NewInterfaceCreationWizardPage interfacePage = new NewInterfaceCreationWizardPage(interfaceDialog);
+		interfacePage.setName(INTERFACE_NAME + "2");
+		interfacePage.setPackage(PACKAGE);
+		interfacePage.setSourceFolder(SOURCE_FOLDER_NAME);
+
+		// try to add a non-existing extended interface
+		boolean exceptionThrown = false;
+		try {
+			interfacePage.addExtendedInterface("nonexisting.interface.name");
+		} catch (RedDeerException e) {
+			exceptionThrown = true;
+		}
+		assertTrue("RedDeer exception was not thrown when trying to add a non-existing extended interface.",
+				exceptionThrown);
+
+		// try to add an extended interface, that is not fully specified by its name
+		exceptionThrown = false;
+		try {
+			interfacePage.addExtendedInterface("Acl");
+		} catch (RedDeerException e) {
+			exceptionThrown = true;
+		}
+		assertTrue(
+				"RedDeer exception was not thrown when trying to add an extended interface, that is not fully specified by its name.",
+				exceptionThrown);
+
+		// try to remove an extended interface, that is not on the list
+		exceptionThrown = false;
+		try {
+			interfacePage.removeExtendedInterface("javax.accessibility.AccessibleAction");
+		} catch (RedDeerException e) {
+			exceptionThrown = true;
+		}
+		assertTrue(
+				"RedDeer exception was not thrown when trying to remove an extended interface, that is not on the list.",
+				exceptionThrown);
+
+		// try to remove an added extended interface, that was not fully specified
+		exceptionThrown = false;
+		try {
+			interfacePage.addExtendedInterface("org.w3c.dom.views.AbstractView");
```

Comment:
Extract this line above the try-catch as it might throw an exception as well.

Reply:
Fixed.